### PR TITLE
[ANOMALY_CLASSIFICATION] Modify Exportable Code

### PR DESCRIPTION
**Pull request by ashwinvaidya17**
# Changes

- Add normalization support to exportable code.

### CODE REVIEW

File: external/anomaly/ote_anomalib/exportable_code/anomaly_classification.py [hunk 1]
```diff
@@ -35,6 +36,11 @@ def parameters(cls):
         parameters.update(
             {
                 "image_threshold": NumericalValue(default_value=0.5, description="Threshold value to locate anomaly"),
```
Comment (djdameln) [inline external/anomaly/ote_anomalib/exportable_code/anomaly_classification.py]: Can we remove the `default_value` here, like for the other stats?

File: external/anomaly/ote_anomalib/exportable_code/anomaly_classification.py [hunk 2]
```diff
@@ -50,10 +56,31 @@ def postprocess(self, outputs: Dict[str, np.ndarray], meta: Dict[str, Any]) -> n
...
+        meta["pixel_threshold"] = self.pixel_threshold  # pylint: disable=no-member
+        meta["image_mean"] = self.image_mean  # pylint: disable=no-member
+        meta["image_std"] = self.image_std  # pylint: disable=no-member
+
+        # standardize pixel scores
+        if "pixel_mean" in meta.keys() and "pixel_std" in meta.keys():
+            anomaly_map = np.log(anomaly_map)
+            anomaly_map = (anomaly_map - meta["pixel_mean"]) / meta["pixel_std"]
+            anomaly_map -= (meta["image_mean"] - meta["pixel_mean"]) / meta["pixel_std"]
+            anomaly_map = norm.cdf(anomaly_map - meta["pixel_threshold"])
```
Comment (samet-akcay) [inline external/anomaly/ote_anomalib/exportable_code/anomaly_classification.py:74]: It would also be an idea to create a function in `anomalib` to perform this operation. This function could then be used both here and in ScoreNormalizationCallback in `anomalib`. I guess it's ok for now, but would be good practice in general since we need to modify things multiple times nowadays.

Comment (ashwinvaidya17) [reply external/anomaly/ote_anomalib/exportable_code/anomaly_classification.py:74]: I agree. My only concern is the torch and numpy inputs. Maybe we can look into a design which adds normalization to forward method during onnx conversion. This way we will directly get normalized anomaly maps without needing to export all the thresholds as well.

Comment (djdameln) [reply external/anomaly/ote_anomalib/exportable_code/anomaly_classification.py:74]: We'll need separate functions for torch and numpy. I don't think including it in the forward method is a good idea. With the current design we can easily disable the normalization and switch between different normalization methods. Also the idea is to keep the forward method as clean as possible and only use it for the model's core functionality. Including the post-processing steps in the forward method will step away from this design and could lead to problems down the road, like we've seen with tiling.

Comment (samet-akcay) [reply external/anomaly/ote_anomalib/exportable_code/anomaly_classification.py:74]: yeah. Plus, we already have this in a normalization callback, which limits the use of this in a forward method. 

What I meant though is we could still wrap this in a function to use it in both the callback and here. I'm currently looking at the implementation. Looks like the only difference is `torch.log` vs `np.log`. We could have an argument to properly check and assign this.
